### PR TITLE
Em 3092 expose the air manager status api

### PR DESCRIPTION
**Pull request by suppomi**
The test contains jsonscheme based API check and heatable object based reaction test.
Good progress of this branch is that now we have print core/build plate heat up function in it.

### CODE REVIEW

File: tests/api/steps/airmanager.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    except IndexError:
+        LOGGING.info("airmanager@0.service is not loaded = NOT CONNECTED")
+        airmanager_api_response = Ultimaker3(context.printer_address, 'automation').get('/api/v1/airmanager')
+        airmanager_status = json.loads(airmanager_api_response.content)["status"]
+        LOGGING.info(f"response for the /api/v1/airmanager was as follows: {airmanager_api_response.content}")
+        assert airmanager_api_response.status_code == 200, "airmanager api does not reply status_code 200"
+        assert airmanager_status == 'unavailable', "airmanager api does not reply 'unavailable' state"
+        # raise AssertionError
+
+    # check if air manager is installed and registry entry is existing on a certain file.'
```
Comment (DanielUltimaker) [inline tests/api/steps/airmanager.py]: Could be the following lines be a method in a helper class? I think that could be reused in other tests.

File: tests/api/Airmanager.feature [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+     Examples:
+     |heatable_object|temperature|expected_fan_speed|
+     |PrintCore1     |52         |2400              |
+     |PrintCore1     |0          |0                 |
+     |PrintCore2     |38         |0                 |
+     |PrintCore2     |51         |2400              |
+     |PrintCore2     |0          |0                 |
+     |BuildPlate     |51         |2400              |
+     |BuildPlate     |0          |0                 |
```
Comment (DanielUltimaker) [inline tests/api/Airmanager.feature]: Why are you just checking the expected fan speed? Maybe could be nice to check that other results are in the expected range or not null.

File: tests/api/steps/airmanager.py [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+def printer_is_idle_state(context):
+    context.um_api = Ultimaker3(context.printer_address, 'automation')
+    printer_status = context.um_api.get('/api/v1/printer/status').content
+    LOGGING.info(f'printer status from api /api/v1/printer/status : {printer_status}')
+    assert b'idle' in printer_status, 'printer is not in idle state'
+
+
+@step("I wait all the heatable objects to be bellow 50 degree C")
+def setup_precondition(context):
+    # checking current target temperarure and set 0 if not
```
Comment (DanielUltimaker) [inline tests/api/steps/airmanager.py]: These lines can be extracted as a method and be moved to a helper class.

File: tests/api/steps/airmanager.py [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    for i in range(3):
+        uri = HeatableObjects.list_api_temperature[i]
+        if float(context.um_api.get(uri + endpoint).content) != 0:
+            if i < 2:
+                response = PrinterActions.setPrintCoreTemp(context.printer_address, i, 0)
+                assert 'boolean true' in response, f'set target temperature failed for print core {i}.'
+            else:
+                response = PrinterActions.setBuildPlateTemp(context.printer_address, 0)
+                assert 'boolean true' in response, f'set target temperature failed for buildplate.'
+    # wait for the each temperature to bellow 50.0 degree C
```
Comment (DanielUltimaker) [inline tests/api/steps/airmanager.py]: Same as the previous comment.

File: tests/api/steps/airmanager.py [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+    if heatable_object_no < 2:
+        response = PrinterActions.setPrintCoreTemp(context.printer_address, heatable_object_no, int(temperature))
+    else:
+        response = PrinterActions.setBuildPlateTemp(context.printer_address, int(temperature))
+    assert 'boolean true' in response, f'setting target temperature failed for {heatable_object}.'
+
+
+@step("I wait for the temperature reach")
+def wait_for_the_temperature(context):
+    heatable_object_no = HeatableObjects.list_object_name.index(context.heatable_object)
```
Comment (DanielUltimaker) [inline tests/api/steps/airmanager.py]: This step also could be a helper method.

File: tests/api/steps/airmanager.py (hunk 1 above)
Comment (suppomi) [reply tests/api/steps/airmanager.py]: This was done by Satish as pull request :Check if airmanager is connected #21

File: helpers/printer_actions.py [hunk 6]
```diff
@@ -404,3 +404,49 @@ def stopGriffinServices(printer_ip):
...
+        :param printer_ip: IP of the printer to connect.
+        :param hotend_no: 0 or 1.
+        :param target_tmep: integer number, 0 for heater off.
+        """
+        """ shell command
+        # dbus-send --system --type=method_call --print-reply --dest="nl.ultimaker.printer" /nl/ultimaker/printer 
+                 nl.ultimaker.startProcedure string:"SET_HOTEND_TEMPERATURE" dict:string:int32:"hotend_nr",0,"target",50
+        boolean true
+        """
+        service = "nl.ultimaker.printer /nl/ultimaker/printer nl.ultimaker.startProcedure "
```
[the remark this answers is not in the document]
Comment (suppomi) [reply helpers/printer_actions.py]: Done. 
same way as active leveling procedure is used.

File: helpers/printer_actions.py [hunk 7]
```diff
@@ -404,3 +404,49 @@ def stopGriffinServices(printer_ip):
...
+        set the build palte temperature .
+        :param printer_ip: IP of the printer to connect.
+        :param target_tmep: integer number, 0 for heater off.
+        """
+        """ shell command
+        # dbus-send --system --type=method_call --print-reply --dest="nl.ultimaker.printer" /nl/ultimaker/printer 
+                 nl.ultimaker.startProcedure string:"SET_BED_TEMPERATURE" dict:string:int32:"target",50
+        boolean true
+        """
+        service = "nl.ultimaker.printer /nl/ultimaker/printer nl.ultimaker.startProcedure "
```
[the remark this answers is not in the document]
Comment (suppomi) [reply helpers/printer_actions.py]: Done. 
same way as active leveling procedure is used.

File: tests/api/Airmanager.feature (hunk 2 above)
Comment (suppomi) [reply tests/api/Airmanager.feature]: New test scenario is added for the purpose of other results in the API.
Note, API does not expose as many as available in DBus property.

File: tests/api/steps/airmanager.py (hunk 3 above)
Comment (suppomi) [reply tests/api/steps/airmanager.py]: done.

File: tests/api/steps/airmanager.py (hunk 1 above)
Comment (suppomi) [reply tests/api/steps/airmanager.py]: done

**Pull request merged**